subiquity/server/controllers/identity.py

```diff
@@ -85,9 +85,13 @@ def load_autoinstall_data(self, data):
 
     @with_context()
     async def apply_autoinstall_config(self, context=None):
-        if not self.model.user:
-            if 'user-data' not in self.app.autoinstall_config:
-                raise Exception("no identity data provided")
+        if self.model.user:
+            return
+        if 'user-data' in self.app.autoinstall_config:
+            return
+        if self.app.controllers.Filesystem.reset_partition_only:
```

Comment:
This is fine for now, but I have a suspicion we'll need to make this bit more extensible in the future - identity to filesystem coupling is a bit surprising.

Reply:
There is another spelling we can use which looks a bit less arbitrary, take a look?